From 612c6897eadabcabd2b3a711fabc92df2e2aeb01 Mon Sep 17 00:00:00 2001
From: Fabian Meumertzheim <fabian@meumertzhe.im>
Date: Wed, 20 Sep 2023 12:54:41 +0200
Subject: [PATCH] versions: Don't fail on Bazel dev builds

Dev builds of Bazel are assumed to be more recent than any released
version.
---
 docs/versions_doc.md     | 3 +++
 lib/versions.bzl         | 5 +++++
 tests/versions_tests.bzl | 5 +++++
 3 files changed, 13 insertions(+)

diff --git a/lib/versions.bzl b/lib/versions.bzl
index 0209a6fa..87ca73c9 100644
--- a/lib/versions.bzl
+++ b/lib/versions.bzl
@@ -44,6 +44,9 @@ def _parse_bazel_version(bazel_version):
 
     int tuples can be compared directly using binary operators (<, >).
 
+    For a development build of Bazel, this returns an unspecified version tuple
+    that compares higher than any released version.
+
     Args:
       bazel_version: the Bazel version string
 
@@ -52,6 +55,8 @@ def _parse_bazel_version(bazel_version):
     """
 
     version = _extract_version_number(bazel_version)
+    if not version:
+        return (999999, 999999, 999999)
     return tuple([int(n) for n in version.split(".")])
 
 def _is_at_most(threshold, version):
